### PR TITLE
[image-builder-mk3] Use baseserver

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Use baseserver in `image-builder-mk3`. Despite our intended move of image builder to WS clusters, this is a worthwhile change to standardise the underlying server.

The second reason to do this is currently grpc server metrics are not properly registered on image-builder, moving it to base-server fixes that as a nice side-effect (we know it works due to other services).

## Related Issue(s)

Fixes #

## How to test

1. Start a workspace which requires an image build in preview
2. See image gets built

Screenshot 2022-09-27 at 22 06 07

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

**Review by jenting — APPROVED** [review]

File: components/image-builder-api/go/config/config.go [hunk 1]
```diff
@@ -5,62 +5,21 @@
...
-}
-
 type ServiceConfig struct {
 	Orchestrator Configuration  `json:"orchestrator"`
 	RefCache     RefCacheConfig `json:"refCache,omitempty"`
-	Service      Service        `json:"service"`
-	Prometheus   Service        `json:"prometheus"`
-	PProf        PProf          `json:"pprof"`
+
+	Server *baseserver.Configuration
```
Comment (csweichel) [inline components/image-builder-api/go/config/config.go:15]: missing `json:"server"`?

File: components/image-builder-mk3/cmd/run.go [hunk 2]
```diff
@@ -87,52 +56,23 @@ var runCmd = &cobra.Command{
...
-			if err != nil {
-				log.Fatal(err)
-			}
-		}
 
-		err = service.Start(ctx)
+		err = service.RegisterMetrics(srv.MetricsRegistry())
 		if err != nil {
-			log.Fatal(err)
+			log.WithError(err).Fatal("Failed to register metrics.")
```
Comment (csweichel) [inline components/image-builder-mk3/cmd/run.go:62]: nit

```suggestion
			log.WithError(err).Fatal("failed to register metrics.")
```

**Review by csweichel — APPROVED** [review]
Image builder works as expected, metrics are exposed as expected.

A few minor nits, otherwise this LGTM

File: components/image-builder-mk3/cmd/run.go (hunk 2 above)
Comment (easyCZ) [reply components/image-builder-mk3/cmd/run.go:62]: This is the log message which can remain capitalized, the error that it attaches is however lowercased

**Review by Pothulapati — APPROVED** [review]
LGTM! 👀

**Pull request merged**